openapi/src/main/java/io/micronaut/openapi/visitor/AbstractOpenApiVisitor.java

```diff
@@ -1189,42 +1179,168 @@ protected void processJavaxValidationAnnotations(Element element, ClassElement e
+                            ((Optional<Integer>) ann.get("min", Integer.class))
+                                .ifPresent(schemaToBind::setMinLength);
+                            ((Optional<Integer>) ann.get("max", Integer.class))
+                                .ifPresent(schemaToBind::setMaxLength);
+                        }
+                    });
             }
 
-            if (element.isAnnotationPresent(Negative.class)) {
+            if (element.isAnnotationPresent("javax.validation.constraints.Negative$List")
+                || element.isAnnotationPresent("jakarta.validation.constraints.Negative$List")) {
                 schemaToBind.setMaximum(BigDecimal.ZERO);
             }
-            if (element.isAnnotationPresent(NegativeOrZero.class)) {
+            if (element.isAnnotationPresent("javax.validation.constraints.NegativeOrZero$List")
+                || element.isAnnotationPresent("jakarta.validation.constraints.NegativeOrZero$List")) {
                 schemaToBind.setMaximum(BigDecimal.ZERO);
             }
-            if (element.isAnnotationPresent(Positive.class)) {
+            if (element.isAnnotationPresent("javax.validation.constraints.Positive$List")
+                || element.isAnnotationPresent("jakarta.validation.constraints.Positive$List")) {
                 schemaToBind.setMinimum(BigDecimal.ZERO);
             }
-            if (element.isAnnotationPresent(PositiveOrZero.class)) {
+            if (element.isAnnotationPresent("javax.validation.constraints.PositiveOrZero$List")
+                || element.isAnnotationPresent("jakarta.validation.constraints.PositiveOrZero$List")) {
                 schemaToBind.setMinimum(BigDecimal.ZERO);
             }
-            element.getValue(Max.class, BigDecimal.class).ifPresent(schemaToBind::setMaximum);
-            element.getValue(Min.class, BigDecimal.class).ifPresent(schemaToBind::setMinimum);
-            element.getValue(DecimalMax.class, BigDecimal.class).ifPresent(schemaToBind::setMaximum);
-            element.getValue(DecimalMin.class, BigDecimal.class).ifPresent(schemaToBind::setMinimum);
-            if (element.isAnnotationPresent(Email.class)) {
-                schemaToBind.setFormat(PrimitiveType.EMAIL.getCommonName());
-            }
 
-            element.findAnnotation(Pattern.class).flatMap(p -> p.get("regexp", String.class)).ifPresent(schemaToBind::setPattern);
-            element.getValue(Part.class, String.class).ifPresent(schemaToBind::setName);
+            element.findAnnotation("javax.validation.constraints.Min$List")
+                .ifPresent(listAnn -> {
+                    for (AnnotationValue ann : (Set<AnnotationValue>) listAnn.getValues().get("value")) {
+                        ((Optional<BigDecimal>) ann.getValue(BigDecimal.class))
+                            .ifPresent(schemaToBind::setMinimum);
+                    }
+                });
```

Comment:
The code in these lines and the next is pretty much identical except for changing the annotation name and he underlying type (BigDecimal vs String). Refactor this to avoid duplication